### ISSUE TITLE
2x2x2 downsampling issues

### ISSUE THREAD

**Issue by cnzqy1** [seung-lab/igneous#83]
I tried to downsample the volume 2x2x2 with a very simple Python script:

```
from taskqueue import LocalTaskQueue
import igneous.task_creation as tc

src_layer_path = 'file://output'
dest_layer_path = 'file://output2'

with LocalTaskQueue(parallel=8) as tq:
  tasks = tc.create_transfer_tasks(
    src_layer_path, dest_layer_path, 
    chunk_size=(64,64,16), skip_downsamples=True, compress='gzip'
  )
  tq.insert_all(tasks)
  tasks = tc.create_downsampling_tasks(
    dest_layer_path, factor=(2,2,2), compress='gzip', num_mips=1
  )
  tq.insert_all(tasks)

print("Done!")
```

However, after the task finished I have a 4_4_4 folder (mip 0) and an 8_8_4 folder (mip 1), although the contents in the 8_8_4 folder are indeed 2x2x2 downsampled so just the folder name is wrong. In addition, the `info` file generated is also incorrect:

```
{
  "data_type": "uint8",
  "num_channels": 1,
  "scales": [
    {
      "chunk_sizes": [
        [
          64,
          64,
          16
        ]
      ],
      "encoding": "raw",
      "key": "4_4_4",
      "resolution": [
        4,
        4,
        4
      ],
      "size": [
        412,
        914,
        800
      ],
      "voxel_offset": [
        0,
        0,
        0
      ]
    },
    {
      "chunk_sizes": [
        [
          64,
          64,
          16
        ]
      ],
      "encoding": "raw",
      "key": "8_8_4",
      "resolution": [
        8,
        8,
        4
      ],
      "size": [
        206,
        457,
        800
      ],
      "voxel_offset": [
        0,
        0,
        0
      ]
    },
    {
      "chunk_sizes": [
        [
          64,
          64,
          16
        ]
      ],
      "encoding": "raw",
      "key": "16_16_4",
      "resolution": [
        16,
        16,
        4
      ],
      "size": [
        103,
        229,
        800
      ],
      "voxel_offset": [
        0,
        0,
        0
      ]
    },
    {
      "chunk_sizes": [
        [
          64,
          64,
          16
        ]
      ],
      "encoding": "raw",
      "key": "8_8_8",
      "resolution": [
        8,
        8,
        8
      ],
      "size": [
        206,
        457,
        400
      ],
      "voxel_offset": [
        0,
        0,
        0
      ]
    }
  ],
  "type": "image"
}
```

As it contains many non-existent layers. After manually correcting the folder name to 8_8_8 and fixing the `info` file, the volume appears to be correct and can be visualized with Neuroglancer. All of my packages (Igneous, CloudVolume, etc.) are up-to-date as of right now.

**Comment by william-silversmith** [comment]
Thanks for reporting this! I'll have to take a look.

**Comment by cnzqy1** [comment]
I believe I have identified the source of the issue. I split the transfer task and the downsampling task into two scripts, and after running transfer the info file looks like this even with `skip_downsamples=True`:

```
{
  "data_type": "uint8",
  "num_channels": 1,
  "scales": [
    {
      "chunk_sizes": [
        [
          64,
          64,
          16
        ]
      ],
      "encoding": "raw",
      "key": "4_4_4",
      "resolution": [
        4,
        4,
        4
      ],
      "size": [
        412,
        914,
        800
      ],
      "voxel_offset": [
        0,
        0,
        0
      ]
    },
    {
      "chunk_sizes": [
        [
          64,
          64,
          16
        ]
      ],
      "encoding": "raw",
      "key": "8_8_4",
      "resolution": [
        8,
        8,
        4
      ],
      "size": [
        206,
        457,
        800
      ],
      "voxel_offset": [
        0,
        0,
        0
      ]
    },
    {
      "chunk_sizes": [
        [
          64,
          64,
          16
        ]
      ],
      "encoding": "raw",
      "key": "16_16_4",
      "resolution": [
        16,
        16,
        4
      ],
      "size": [
        103,
        229,
        800
      ],
      "voxel_offset": [
        0,
        0,
        0
      ]
    }
  ],
  "type": "image"
}
```

After correcting the `info` file running the downsampling task gives the correct results.

**Comment by william-silversmith** [comment]
Ahh, I see it. In `create_transfer_tasks` the skip_downsamples flag is not respected and `create_downsample_scales` is always called. Thanks for pointing this out!

**Comment by william-silversmith** [comment]
Fixed in 543215ec468bad58515a36b0c8eba7ab759cf510 let me know if that doesn't do it for you. ^_^
